core/shared/platform/windows/win_socket.c

```diff
@@ -54,18 +77,24 @@ os_socket_create(bh_socket_t *sock, bool is_ipv4, bool is_tcp)
     }
 
     if (is_tcp) {
-        *sock = socket(af, SOCK_STREAM, IPPROTO_TCP);
+        (*sock)->raw.socket = socket(af, SOCK_STREAM, IPPROTO_TCP);
     }
     else {
-        *sock = socket(af, SOCK_DGRAM, 0);
+        (*sock)->raw.socket = socket(af, SOCK_DGRAM, 0);
+    }
+
+    if ((*sock)->raw.socket == -1) {
```

Comment:
According to this: https://learn.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-socket#return-value we should check for `INVALID_SOCKET`

Reply:
Also, if it's simple we might want to also set the errno here as well (but if that adds a lot more work, I think it's ok to ignore it for now).

Reply:
Thanks, have changed it to use `INVALID_SOCKET`. I'll leave setting errno for the moment if that's ok - mainly wanted to focus on the filesystem changes in this PR. But it should just be a matter of calling `WSAGetLastError` and mapping the winsock error code to an errno code.